### PR TITLE
EU-118

### PR DESCRIPTION
**Pull request by victor-nuzhniy**

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -56,7 +59,7 @@ class MainSettings(BaseSettings):
     PORT: int = Field(default=8000)
     WORKERS_COUNT: int = Field(default=1)
     TRUSTED_HOSTS: List[str] = Field(default=["*"])
-    DATETIME_FORMAT: str = Field("%Y-%m-%d %H:%M:%S")
+    DATETIME_FORMAT: str = Field("%Y-%m-%d %H-%M-%S")
```
Comment (Kostiantyn-Salnykov) [inline settings.py]: Why did you change whole project datetime format, please refer to ISO-8601:
YYYY-MM-DDTHH:MM:SS+-HHMM

**Review by Kostiantyn-Salnykov — CHANGES_REQUESTED** [review]

File: settings.py (hunk 1 above)
Comment (victor-nuzhniy) [reply settings.py]: It was not good attempt to avoid error with service file creating. Of course I'll fix it now

File: apps/common/file_manager.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                callable(subclass.create) and 
+                hasattr(subclass, 'get') and 
+                callable(subclass.get) and 
+                hasattr(subclass, 'delete') and 
+                callable(subclass.delete) or 
+                NotImplemented)
+
+    @abc.abstractmethod
+    def create(self, path: str, file_name: str, rendered_template: DocxTemplate) -> str:
+        """Create new file"""
```
Comment (anton-serozhechkin) [inline apps/common/file_manager.py]: please delete useless comments from this file

File: apps/common/file_manager.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+class FileManagerInterface(metaclass=abc.ABCMeta):
+    @classmethod
+    def __subclasshook__(cls, subclass):
+        return (hasattr(subclass, 'create') and 
+                callable(subclass.create) and 
+                hasattr(subclass, 'get') and 
+                callable(subclass.get) and 
+                hasattr(subclass, 'delete') and 
+                callable(subclass.delete) or
```
Comment (anton-serozhechkin) [inline apps/common/file_manager.py]: please also add render method

File: apps/services/handlers.py [hunk 4]
```diff
@@ -181,6 +182,22 @@ async def read_request_details(
...
 
+    async def read_service_document(
+            self,
+            *,
+            request: Request,
+            university_id: int,
+            user_document_id: int,
+            user: UserOut,
+            session: AsyncSession):
+        if user.university_id != university_id:
```
Comment (anton-serozhechkin) [inline apps/services/handlers.py]: i think it's useless code for now, bcs we didn't start to implement user roles/actions schema

File: apps/services/handlers.py [hunk 5]
```diff
@@ -181,6 +182,22 @@ async def read_request_details(
                 "user_request_id": user_request_id
             })
 
+    async def read_service_document(
```
Comment (anton-serozhechkin) [inline apps/services/handlers.py]: why service? should be read_user_document

**Review by anton-serozhechkin — CHANGES_REQUESTED** [review]

File: apps/common/file_manager.py (hunk 2 above)
Comment (victor-nuzhniy) [reply apps/common/file_manager.py]: ok

File: apps/common/file_manager.py (hunk 3 above)
Comment (victor-nuzhniy) [reply apps/common/file_manager.py]: ok

File: apps/services/handlers.py (hunk 4 above)
Comment (victor-nuzhniy) [reply apps/services/handlers.py]: ok

File: apps/services/handlers.py (hunk 5 above)
Comment (victor-nuzhniy) [reply apps/services/handlers.py]: ok

File: apps/common/file_manager.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import abc
+
+from docxtpl import DocxTemplate
+from pathlib import Path
```
Comment (anton-serozhechkin) [inline apps/common/file_manager.py]: pathlib is included to python core library, so it should be at the same import level as abc lib

File: apps/common/file_manager.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    @abc.abstractmethod
+    def render(self, path: str, full_file_path: str, data: dict) -> DocxTemplate:
+        raise NotImplementedError
+
+
+class FileManagerLocal(FileManagerInterface):
+    def create(self, path: str, file_name: str, rendered_template: DocxTemplate) -> str:
+        path_to_storage = Path(path) / file_name
+        rendered_template.save(path_to_storage)
+        return str(path_to_storage)
```
Comment (anton-serozhechkin) [inline apps/common/file_manager.py]: a u sure that we need wrap it to str?

File: apps/services/routers.py [hunk 8]
```diff
@@ -261,6 +262,33 @@ async def read_request_details(
     }
 
 
+@services_router.get("/{university_id}/service-document/{user_document_id}",
```
Comment (anton-serozhechkin) [inline apps/services/routers.py]: why service document? should be user-document

File: apps/services/routers.py [hunk 9]
```diff
@@ -261,6 +262,33 @@ async def read_request_details(
...
+                     name="read_service_document",
```
Comment (anton-serozhechkin) [inline apps/services/routers.py]: read_user_document

File: apps/services/routers.py [hunk 10]
```diff
@@ -261,6 +262,33 @@ async def read_request_details(
...
+                     response_class=StreamingResponse,
+                     summary="Read service document",
```
Comment (anton-serozhechkin) [inline apps/services/routers.py]: user document

File: apps/services/routers.py [hunk 11]
```diff
@@ -261,6 +262,33 @@ async def read_request_details(
...
+                     responses={200: {
+                         "description": "Successful get service document response",
```
Comment (anton-serozhechkin) [inline apps/services/routers.py]: user document

**Review by anton-serozhechkin — DISMISSED** [review]

**Review by anton-serozhechkin — APPROVED** [review]

File: apps/services/handlers.py [hunk 12]
```diff
@@ -210,7 +223,7 @@ async def count_hostel_accommodation_cost(
 
     @classmethod
     def calculate_difference_between_dates_in_months(cls, end_date: date, start_date: date) -> int:
-        return end_date.month - start_date.month + 12 * (end_date.year - start_date.year)
+        return (end_date.month - start_date.month + 12 * (end_date.year - start_date.year)) + 1
```
Comment (YehorVoytenko) [inline apps/services/handlers.py:226]: why changed this method? @anton-serozhechkin that issue we tried to fix?

**Comment by YehorVoitenko** [comment]
Approved

**Review by YehorVoytenko — APPROVED** [review]
Approved

**Review by anton-serozhechkin — APPROVED** [review]
